docs/source-2.0/spec/constraint-traits.rst

```diff
@@ -12,6 +12,20 @@ impact the types signatures of generated code.
        that also functions like a constraint.
 
 
+Constraint trait enforcement
+============================
+
+Constraint traits SHOULD be enforced after deserializing input. For example,
+when a server deserializes a request from a client, the server SHOULD enforce
+any defined constraint traits and reject the request if appropriate.
+
+Constraint traits SHOULD NOT be enforced when serializing shapes or when
+deserializing output. For example, when returning a response from a server
+to a client, failing to serialize a response due to a constraint trait
+violation would prevent a client from observing a state change of the server
+and would provide no real recourse for the client or server to recover.
+
+
```

Comment:
A link or note to [this section](https://smithy.io/2.0/guides/building-codegen/mapping-shapes-to-languages.html#why-don-t-we-validate-constraint-traits-on-the-client) would be helpful here, some users get confused as of why they can build invalid structs.

Reply:
I try not to link from the normative spec out of the spec to non-normative content